### PR TITLE
Andrewrutter/kkb 41 event signup frontend

### PR DESCRIPTION
**Pull request by andlrutt**
This is the frontend component for event sign-ups. Collects name, email, and phone number, and requires that volunteers acknowledge the waiver. The waiver has not been linked, but it'll be easy enough to add a `to="/path"` to the Link element once that is established. Down the line, we'll need to add functionality to the sign-up button to check each field to make sure it has valid text, as currently the button currently only logs that it was pressed to the console. Also added a `lightGray` color to be used for input fields across the site. Below is what the component looks like.

EDIT: Since I submitted this PR, Kemal pointed out some areas in which this component could be improved, and in the process I noticed some areas as well. I have since made it so the component is converted into a vertical layout if the window is too small, prevented the sign-up button from being clicked unless all required data is entered, and made a function that takes the form data and makes a Volunteer object (should be VERY easy to create a volunteer from this, just implement a post request). Along with these, I made several small changes that slightly improve the functionality/appearance of the component and readability of the file. The new (only VERY slightly different) layout is below, along with what it looks like when the window is too small for the default layout (2 images).

One thing I noted is that the phone number pretty much has to be taken in as a string, so I was wondering how we would store that. Would we want to store the raw string which might have dashes in it, or maybe parse all the numbers in the string? I'd like to know what you all think.

img1
img2
img

### CODE REVIEW

File: src/components/EventForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            border: "none",
+            color: colors.white,
+            borderRadius: "6px",
+            fontSize: "20px",
+            fontFamily: "Roboto",
+        },
+    })
+);
+
+export default function EventsForm() {
```
Comment (kemalfidan) [inline src/components/EventForm/index.tsx]: Can you name this `EventForm` to be consistent with the filename? I actually like `EventSignUp` better.

File: src/components/EventForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        container: {
+            display: "flex",
+            textAlign: "center",
+            width: "750px",
+            borderStyle: "solid",
+            borderWidth: "2px",
+            borderColor: colors.green,
```
Comment (kemalfidan) [inline src/components/EventForm/index.tsx]: Can you use the primary color from `theme` here? Should be `theme.palette.primary.main`.

File: src/components/EventForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            height: "700px",
+            paddingLeft: "0",
+        },
+
+        leftWrapper: {
+            backgroundColor: colors.green,
```
Comment (kemalfidan) [inline src/components/EventForm/index.tsx]: `theme.palette.primary.main`

File: src/components/EventForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        checkboxText: {
+            width: "240px",
+            paddingTop: "10px",
+        },
+
+        button: {
+            marginTop: "25px",
+            marginLeft: "140px",
+            padding: "15px",
+            backgroundColor: colors.green,
```
Comment (kemalfidan) [inline src/components/EventForm/index.tsx]: Same as above

**Review by kemalfidan — DISMISSED** [review]
I noticed a few things:
1. The component isn't really responsive. It looks nice on a large screen but when I make the screen size smaller, the display doesn't adjust to that screen. To fix this, can you add a media query to stack the signup form rather than put it side by side. I'll try to mock this up quickly on Figma.
2. Is it possible to get a `Volunteer` variable of the data in the form? There's different ways to set this up but I think [this](https://github.com/hack4impact-utk/mindversity-website/blob/develop/components/CreateJournalEntry/index.tsx) is probably a good reference. Edit: [You can also do it by references](https://github.com/hack4impact-utk/mindversity-website/blob/develop/pages/portal/resources/create.tsx), which might use fewer code changes.

File: src/components/EventSignUp/index.tsx [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+            },
+        },
+
+        leftWrapper: {
+            backgroundColor: theme.palette.primary.main,
+            color: colors.white,
+            paddingTop: "220px",
+            height: "700px",
+            [theme.breakpoints.between(0, 720)]: {
+                width: "371px",
```
Comment (kemalfidan) [inline src/components/EventSignUp/index.tsx]: Can you decrease the height here? I feel like it's a lot to scroll through when looking on mobile. A short header like on the Figma would look better I think. Feel free to let me know what you/others think!

File: src/components/EventSignUp/index.tsx [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+    // on sign-up button click
+    const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+
+        // creates Volunteer object
+        const volunteer: Volunteer = {
+            name: firstName.current!.value + " " + lastName.current!.value,
+            email: email.current!.value,
+            phone: phoneNumber.current!.value,
```
Comment (kemalfidan) [inline src/components/EventSignUp/index.tsx]: phone number might be optional so I think this should be `phoneNumber.current?.value`

**Review by kemalfidan — COMMENTED** [review]
The changes look great! Thanks for handling that. Are you still using the waiver symbol image under public? If not do you mind removing that? I know Java has a mask formatted for formatting the input - [this](https://www.npmjs.com/package/react-input-mask) seems like it might be useful for what we're looking for. A simple (XXX)XXX-XXXX or XXX-XXX-XXXX is what we're looking for.

File: src/components/EventSignUp/index.tsx (hunk 5 above)
Comment (andlrutt) [reply src/components/EventSignUp/index.tsx]: Ah, didn't catch that mockup on Figma, thanks!

**Review by zaviermiller — APPROVED** [review]

**Review by rossketron — APPROVED** [review]
Looks great!

**Pull request merged**